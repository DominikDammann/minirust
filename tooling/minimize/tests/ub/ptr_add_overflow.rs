fn main() {
    let x = &[0i32; 2];
    let x = std::ptr::from_ref(&x[0]).wrapping_add(1);
    // Will be equal to -4isize when multiplied be the size (4) -- and that step does not itself overflow.
    let offset = !0usize >> 2;
    // However, the usize-to-isize cast is lossy and hence this should be UB.
    // Or put differently, -4isize as usize is out-of-bounds.
    unsafe { x.add(offset).read() };
}
