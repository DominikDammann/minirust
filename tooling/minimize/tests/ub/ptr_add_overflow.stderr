fatal error: UB: dereferencing pointer outside the bounds of its allocation
